src/Orleans/Messaging/SocketManager.cs

```diff
@@ -65,9 +67,16 @@ private Socket SendingSocketCreator(IPEndPoint target)
                 s.NoDelay = true;
                 WriteConnectionPreemble(s, Constants.SiloDirectConnectionId); // Identifies this client as a direct silo-to-silo socket
                 // Start an asynch receive off of the socket to detect closure
-                var foo = new byte[4];
-                s.BeginReceive(foo, 0, 1, SocketFlags.None, ReceiveCallback,
```

Comment:
sure, I wasn't even sure whether to include this commit... Nevertheless, it IS the only usage of BeginReceive in the Orleans project. The others are in OrleansRuntime, which I did not try to port to CoreCLR yet.
I could not include this commit if you have any concerns, I don't really care that much about this one.